### DISCUSSION TITLE
Test hangs forever when using AsyncClient.stream() on a SSE endpoint

### DISCUSSION THREAD

**Discussion by havardthom** [encode/httpx#1787] [Potential Issue]
Hi, 
I have an SSE endpoint created with FastAPI + sse-starlette + aioredis. The SSE endpoint is subscribed to a redis channel and returns an event source response to the connected client whenever a message is published to the channel.

I'm trying to test this SSE endpoint using HTTPX AsyncClient but the test hangs forever when I try to run it. I suspect it is because the test code and the server code is running on the same asyncio loop and the thread is blocked by the SSE endpoint code waiting for a message from redis. Has anyone run into this problem before, is there a smart way to run the test code in a separate thread from the server code so it is not blocked by a running SSE loop?

Some pseudo code for my test to show what im trying
```
async def test_sse_endpoint(self, client):
    headers = {"Cache-Control": "no-cache", "Accept": "text/event-stream"}

    # Test gets blocked at this point and hangs forever
    async with client.stream("GET", "sse_url", headers=headers) as response:

        # Post something -> publishes a message to redis -> message should be picked up by SSE endpoint
        await client.post(...)  
        
        # Check that I get some response
        async for line in response.aiter_lines():
            print(line)
```

(Also found a similar issue which could be related: https://github.com/tiangolo/fastapi/issues/2006)

**Comment by havardthom** [comment]
Similar to https://github.com/tiangolo/fastapi/issues/2006 I got my test working with https://pypi.org/project/async-asgi-testclient/

So I believe there is some issue with AsyncClient.stream() which causes the test to hang forever.

#### Working pseudo test:
```
from async_asgi_testclient import TestClient

async def test_sse_endpoint(self, app: FastAPI):
    headers = {"Cache-Control": "no-cache", "Accept": "text/event-stream"}
    
    async with TestClient(app) as client:
        response = await client.get("sse_url", headers=headers, stream=True)

         # Post something -> publishes a message to redis -> message should be picked up by SSE endpoint
        await client.post(...)

        # Check that I get some response
        async for line in response.iter_content(chunk_size=1024):
            print(line)
```

**Reply by william-silversmith** [reply]
I wonder if this issue is the same as mine: https://github.com/encode/httpx/discussions/1796

**Comment by havardthom** [comment]
Wondering if this will be fixed by https://github.com/encode/httpx/pull/998?
